src/main/webapp/js/components/widgets/Widget.js

```diff
@@ -659,6 +659,9 @@ define(function (require) {
                             if (event.originalEvent &&
                                 $(event.originalEvent.target).closest(".ui-dialog-titlebar-close").length) {
                                 that.destroy();
+                                if(that.reactReference !== undefined) {
+                                    that.reactReference.closeHandler();
```

Comment:
what does closeHandler mean?

Reply:
**From VFBMain.js** (762-766)
`        if((this.state.tutorialWidgetVisible == true) && (this.tutorialRender == undefined)) {
            this.tutorialRender = <TutorialWidget tutorialHandler={this.tutorialHandler} ref="tutorialWidgetRef" />
        } else if((this.state.tutorialWidgetVisible == true) && (this.tutorialRender != undefined)){
            this.refs.tutorialWidgetRef.refs.tutorialRef.open(true);
        }
`
**From VFBMain.js** (710-716)
`    tutorialHandler() {
        if(this.state.tutorialWidgetVisible == true) {
            this.setState({
                tutorialWidgetVisible: false
            });
        }
    }
`
**From TutorialWidget.js** (20-23)
`    closeHandler() {
        console.log("tutorial close handler called");
        this.props.tutorialHandler();
    }
`

I am using this method/props to trigger a set state on VFBMain and detect when the tutorial is present or not.
Same principle will be used for the new term info and the stack viewer when these will move to use the widget capability.
I think I can probably remove this but I thought in the long term will help the transition from jquery to react and minimise the changes required on the VFBMain since the logic and the handler will be already there.

Reply:
Errata corrige, this was used initially when I tried to implement the term info widget and adapt it to use the widget capability, not needed anymore apparently.

Reply:
Errata corrige, this is used for the old term info component into the react refactoring, this will go away when the new term info component will be ready, since then I will keep this in the branch.